src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,793 @@
+      // or maximum size
+      if (newRestrictedInlineSize === dialogRect.width) {
+        return;
+      }
+
+      // - - - - - - - - - - - - - DOM write operations - - - - - - - - - - - - -
+      // By resizing the start edge the control is translated to improve the UX
+      if (direction === "start") {
+        this.#draggedDistanceXForResize -= currentDraggedDistanceX;
+
+        setProperty(this.el, DIALOG_DRAGGED_X, this.#draggedDistanceXForResize);
+      }
+
+      setProperty(this.el, DIALOG_INLINE_SIZE, newRestrictedInlineSize);
+    }
+  } as const;
+
+  #resizeEdgesAndCornersDictionary: {
+    [key in ChDialogResizeElement]: (dialogRect: DOMRect) => void;
+  } = {
+    "block-start": dialogRect =>
+      this.#resizeByDirectionDictionary.block(dialogRect, "start"),
+
+    "block-end": dialogRect =>
+      this.#resizeByDirectionDictionary.block(dialogRect, "end"),
+
+    "inline-start": dialogRect =>
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start"),
+
+    "inline-end": dialogRect =>
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end"),
+
+    "block-start-inline-start": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "start");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start");
+    },
+
+    "block-start-inline-end": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "start");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end");
+    },
+
+    "block-end-inline-start": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "end");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start");
+    },
+
+    "block-end-inline-end": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "end");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end");
+    }
+  };
+
+  // Refs
+  #resizeLayer: HTMLDivElement;
+  #dialogRef: HTMLDialogElement;
+
+  @Element() el: HTMLChDialogElement;
+
+  @State() dragging = false;
+  @State() relativeDialog = false;
+  @State() resizing = false;
+
+  /**
+   * Specifies the drag behavior of the dialog.
+   * If `allowDrag === "header"`, a slot with the `"header"` name will be
+   * available to place the header content.
+   */
+  @Prop() readonly allowDrag: "box" | "header" | "no" = "box";
+
+  /**
+   * Specifies whether the dialog is hidden or visible.
+   */
+  // eslint-disable-next-line @stencil-community/ban-default-true
+  @Prop({ mutable: true, reflect: true }) hidden = false;
+  @Watch("hidden")
+  handleHiddenChange(hidden: boolean) {
+    // Schedule update for watchers
+    this.#checkBorderSizeWatcher = true;
+    this.#checkPositionWatcher = true;
+
+    // Update the dialog visualization
+    if (hidden) {
+      this.#dialogRef.close();
+    } else {
+      this.#showMethod();
+    }
+  }
+
+  /**
+   * Specifies whether the dialog is a modal or not.
```

Comment:
Nit: Improve this description. Specify what happens when the dialog is modal and what happens when it is not.